### PR TITLE
Bug 1794077/1780350/1820404 Persist Readerview position

### PR DESCRIPTION
**Pull request by jackyzy823**
1.  Keep Readerview's position between active and inactive
2.  Persiste Readerview's position between restart fenix

Note:  Value of position (scrollY) is accurate, however after fenix restart , the Readerview may not remain in same place ( a bit above). This may be caused by re-generating DOM. There's no such problem when toggling between active and inactive.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
4. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
5. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
6. Click on `View task in Taskcluster` in the new `DETAILS` section.
7. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1820404

### CODE REVIEW

**Comment by Amejia481** [comment]
@jackyzy823 👋🏽 ,
Thanks for the patch!

I think we can't have multiple bug numbers in the commit message e.g `Bug 1794077/1780350`, I think we should target one, you could mention in the main bug the other bugs will be address when the main one lands.

**Comment by jackyzy823** [comment]
@Amejia481  updated.

File: android-components/components/browser/state/src/main/java/mozilla/components/browser/state/state/ReaderState.kt [hunk 1]
```diff
@@ -24,4 +24,5 @@ data class ReaderState(
     val connectRequired: Boolean = false,
     val baseUrl: String? = null,
     val activeUrl: String? = null,
+    val scrollY: Int? = null,
```
Comment (Amejia481) [inline android-components/components/browser/state/src/main/java/mozilla/components/browser/state/state/ReaderState.kt:27]: Nit: Lets add docs for this new property :)

**Review by Amejia481 — APPROVED** [review]
This looks good!
Just a small nit :)

**Pull request merged**